datanode/sqlstore/markets.go

```diff
@@ -21,12 +21,58 @@ import (
 	"code.vegaprotocol.io/vega/datanode/metrics"
 	v2 "code.vegaprotocol.io/vega/protos/data-node/api/v2"
 	"github.com/georgysavva/scany/pgxscan"
+	"golang.org/x/exp/maps"
 )
 
+type cacheKey struct {
+	forwardOffset  int32
+	backwardOffset int32
+	forwardCursor  string
+	backwardCursor string
+	newestFirst    bool
+	marketID       string
+	includeSettled bool
+}
+
+type cacheValue struct {
+	markets  []entities.Market
+	pageInfo entities.PageInfo
+}
+
+func newCacheKey(marketID string, pagination entities.CursorPagination, includeSettled bool) cacheKey {
+	k := cacheKey{
+		marketID:       marketID,
+		newestFirst:    pagination.NewestFirst,
+		includeSettled: includeSettled,
+	}
+
+	if pagination.Forward != nil {
+		if pagination.Forward.Limit != nil {
+			k.forwardOffset = *pagination.Forward.Limit
```

Comment:
Question: Given the limits are part of the cache key, could I as a naughty person fill up the cache and the data-node's memory just by sending in the same valid query over and over again, but just with +1 on the forward limit from `0 -> infinity`?